src/ate/test_programs/dut_lib/dut_lib.cc

```diff
@@ -13,19 +13,20 @@ namespace provisioning {
 namespace test_programs {
 
 extern "C" {
-void* OtLibFpgaInit(const char* fpga, const char* fpga_bitstream);
-void* OtLibLoadSramElf(void* transport, const char* openocd, const char* elf);
+void* OtLibFpgaTransportInit(const char* fpga);
+void OtLibFpgaLoadBitstream(void* transport, const char* fpga_bitstream);
+void OtLibLoadSramElf(void* transport, const char* openocd, const char* elf);
 }
 
-std::unique_ptr<DutLib> DutLib::Create(void) {
-  return absl::make_unique<DutLib>();
+DutLib::DutLib(void* transport) : transport_(transport) {};
```

Comment:
Because this is so simple, you could put this constructor inline in the header files.